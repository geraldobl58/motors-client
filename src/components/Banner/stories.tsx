import { Story, Meta } from '@storybook/react/types-6-0'
import Banner, { BannerProps } from '.'

export default {
  title: 'Banner',
  component: Banner,
  args: {
    img:
      'https://www.autoo.com.br/fotos/2021/11/1280_960/ranger2_26112021_68920_1280_960.jpg',
    title: 'NOVA RANGER 2023',
    subtitle: '<p>V6 TURBODIESEL</p>',
    buttonLabel: 'Ver Mais',
    buttonLink: '/vehicles/nova-ranger'
  }
} as Meta

export const Basic: Story<BannerProps> = (args) => <Banner {...args} />
