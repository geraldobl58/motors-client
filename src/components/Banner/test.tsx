import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Banner from '.'

const props = {
  img:
    'https://www.autoo.com.br/fotos/2021/11/1280_960/ranger2_26112021_68920_1280_960.jpg',
  title: 'NOVA RANGER 2023',
  subtitle: '<p>V6 TURBODIESEL</p>',
  buttonLabel: 'Ver Mais',
  buttonLink: '/vehicles/nova-ranger'
}

describe('<Banner />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Banner {...props} />)

    expect(
      screen.getByRole('heading', { name: /nova ranger 2023/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /turbodiesel/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('img', { name: /ova ranger/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
