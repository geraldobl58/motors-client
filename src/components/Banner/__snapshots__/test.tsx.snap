// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render correctly 1`] = `
.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #FAFAFA;
  background: linear-gradient(180deg,#34abd0 0%,#2d8dab 50%);
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
}

.c5:hover {
  background: linear-gradient(180deg,#2d8dab 0%,#34abd0 50%);
}

.c0 {
  position: relative;
}

.c1 {
  width: 100%;
  height: 23rem;
  background-color: #F3F3F3;
  background-image: url(https://www.autoo.com.br/fotos/2021/11/1280_960/ranger2_26112021_68920_1280_960.jpg);
  background-position: center center;
  background-size: cover;
}

.c2 {
  width: 100%;
  background-color: rgba(0,0,0,0.7);
  padding: 2.4rem;
}

.c3 {
  font-size: 1.8rem;
  font-weight: 700;
  color: #FAFAFA;
}

.c4 {
  color: #FAFAFA;
  font-size: 1.4rem;
  font-weight: 400;
  margin-bottom: 1.6rem;
}

.c4 strong {
  color: #363636;
  font-weight: 700;
}

@media (min-width:768px) {
  .c0 {
    box-shadow: 0 0.4rem 0.5rem 0 rgba(0,0,0,0.2);
  }
}

@media (min-width:768px) {
  .c1 {
    height: 58rem;
  }
}

@media (min-width:768px) {
  .c2 {
    border-radius: 0 0 0.4rem 0.4rem;
    padding: 4.0rem;
    position: absolute;
    bottom: 0;
    left: 0;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 1.8rem;
  }
}

<main
  class="c0"
>
  <div
    aria-label="NOVA RANGER 2023"
    class="c1"
    role="img"
    src="https://www.autoo.com.br/fotos/2021/11/1280_960/ranger2_26112021_68920_1280_960.jpg"
  />
  <div
    class="c2"
  >
    <h2
      class="c3"
    >
      NOVA RANGER 2023
    </h2>
    <h3
      class="c4"
    >
      <p>
        V6 TURBODIESEL
      </p>
    </h3>
    <a
      class="c5"
      href="/vehicles/nova-ranger"
    >
      <span>
        Ver Mais
      </span>
    </a>
  </div>
</main>
`;
